.github/workflows/ci-extern-libs.yaml

```diff
@@ -0,0 +1,269 @@
+name: Build External Libraries
+
+######################################################################
+# Ubuntu testing some libraries.
+# We are particularly interested in libraries that are heavily using
+# dependent types, that are prone to find bugs and regressions in the
+# compiler.
+######################################################################
+
+on:
+  # Copied from the ci-idris2.yml file
+  push:
+    branches:
+      - '*'
+    tags:
+      - '*'
+    paths-ignore:
+      - 'docs/**'
+      - 'icons/**'
+      - 'Release/**'
+      - '**.md'
+      - 'CONTRIBUTORS'
+      - 'LICENSE'
+      - 'CHANGELOG.md'
+      - '.github/workflows/ci-bootstrap.yml'
+      - '.github/workflows/ci-lint.yml'
+      - '.github/workflows/ci-sphinx.yml'
+      - '.github/workflows/ci-super-linter.yml'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - 'docs/**'
+      - 'icons/**'
+      - 'Release/**'
+      - '**.md'
+      - 'CONTRIBUTORS'
+      - 'LICENSE'
+      - '.github/workflows/ci-bootstrap.yml'
+      - '.github/workflows/ci-lint.yml'
+      - '.github/workflows/ci-sphinx.yml'
+      - '.github/workflows/ci-super-linter.yml'
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  IDRIS2_MINIMUM_COMPAT_VERSION: 0.6.0
+
+jobs:
+
+  initialise:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@v3
+        with:
+          # for pull_request so we can do HEAD^2
+          fetch-depth: 2
+
+      - name: Get commit message
+        id: get_commit_message
+        run: |
+          if   [[ '${{ github.event_name }}' == 'push' ]]; then
+            echo "commit_message=$(git log --format=%B -n 1 HEAD | tr '\n' ' ')" >> "$GITHUB_OUTPUT"
+          elif [[ '${{ github.event_name }}' == 'pull_request' ]]; then
+            echo "commit_message=$(git log --format=%B -n 1 HEAD^2 | tr '\n' ' ')" >> "$GITHUB_OUTPUT"
+          fi
+      - name: Download Idris2 Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: ubuntu-installed-idris2-${{ env.IDRIS2_MINIMUM_COMPAT_VERSION }}-chez
+          path: ~/.idris2/
+      - name: Install build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y chezscheme
+          echo "$HOME/.idris2/bin" >> "$GITHUB_PATH"
+          chmod +x "$HOME/.idris2/bin/idris2" "$HOME/.idris2/bin/idris2_app/"*
+      - name: Build self-hosted from previous version
+        run: make all IDRIS2_BOOT="idris2 -Werror" && make install
```

Comment:
The annoying thing with the decoupling is that we are duplicating work by re-building idris here.

Reply:
If we accept `pack` as a dependency, we could just move it all to depend on building `pack`?...  : )

Reply:
What's wrong with the monolithic file?

Reply:
We can save built and installed state of the compiler between jobs. I do it all the time in [my stuff](https://github.com/buzden/deptycheck/blob/7c3a4884b794e60e1edb9a1f052cbbd304421c67/.github/workflows/ci-deptycheck.yml#L100-L106)

Reply:
Doesn't that mean we'd run an old version of the compiler?

Reply:
> Doesn't that mean we'd run an old version of the compiler?

If we save the state after building and installing the new version, we would restore new version too. So, no, not really

Reply:
But the point is that because we're not building in a monolithic file anymore then
we're not ensuring that this build happens before the library testing, are we?

Reply:
> But the point is that because we're not building in a monolithic file anymore then we're not ensuring that this build happens before the library testing, are we?

If I understand your complaint correctly, we can assure this with several files too, by setting dependency to main build jobs and by requiring a built artifact from them.

But if the question is in the count of files, I don't see the **need** for several ones

Reply:
The reason to separate things out into a different file was for clarity: currently, if `ci-idris2` fails, that (to me) means that the Idris2 build failed. But it turns out it could be a number of non-idris2-build related things like katla or collie not compiling. Hence, creating a new CI job called "external-libs".

The work duplication is true, but these jobs (afaik) run in parallel so I'd only consider it semi-wasted work. I'd rather not have this job depend on `ci-idris2` since that would defeat the purpose of concurrency, from my point of view.  
I copied the quick-check and self-host-previous because they seemed like sensible checks before building the external libs. However, from a search through the CI documentation, it seems it is possible to change this such that the `ci-idris2` job cancels this one if it fails, thereby removing the duplicate work while maintaining the separate files.

Thoughts?  : )

Reply:
> since that would defeat the purpose of concurrency, from my point of view.

You can see the dependency graph on any build e.g. https://github.com/idris-lang/Idris2/actions/runs/6089530244?pr=3063

We are already building the external libs concurrently with the ones testing the compiler can self-host with various backends.

The point of having `quick-check` as a universal dependency is that we do not try to build everything else unless we are sure the basic build already works. This helps with concurrency across PRs as we are rate limited in the number of builds we can have running at the same time.

Reply:
> I would rather not have duplicated work at all.

I personally fully agree. Even for the sake of concurrency, I think, it's better to depend on a job rather than to repeat the prefix work

Reply:
Fair enough, I have merged the files back into one to avoid work duplication. I can revert the rename if need be?  : )